src/scripts/scenes/xr-scene.js

```diff
@@ -3,6 +3,7 @@ import { Scene } from 'three';
 export class XrScene {
   scene = new Scene();
   isActive = true;
+  state = {};
```

Comment:
What about a getter for this? We might not want to update this variable for every step of execution, but I like your idea of each scene implementing a way to see the state for when it needs to be saved :)